pkgs/development/misc/resholve/resholve-package.nix

```diff
@@ -0,0 +1,128 @@
+{ stdenv, lib, resholve }:
+
+{ pname
+, src
+, version
+, passthru ? { }
+  /* Each "solution" (k=v pair) describes one resholve invocation.
+   * See 'man resholve' for option details.
+   *
+   * Example:
+   * {
+   *   # name the script(s) for overrides & error messages
+   *   shortname = {
+   *     # required
+   *     # $out-relative paths to try resolving
+   *     scripts = [ "bin/shunit2" ];
+   *     # packages to resolve executables from
+   *     inputs = [ coreutils gnused gnugrep findutils ];
+   *     # path for shebang, or 'none' to omit shebang
+   *     interpreter = "${bash}/bin/bash";
+   *
+   *     # optional
+   *     # to translate fake/fix/keep directives, convert:
+   *     # - `key:value1;value2` to `key: [ value1 value2 ];`
+   *     # - `value` to `value = true;`
+   *     fake = { fake directives };
+   *     fix = { fix directives };
+   *     keep = { keep directives };
+   *     # file to inject before first code-line of script
+   *     prologue = file;
+   *     # file to inject after last code-line of script
+   *     epilogue = file;
+   *     # extra command-line flags passed to resholve; generally this API
+   *     # should align with what resholve supports, but flags may help if
+   *     # you need to override the version of resholve.
+   *     flags = [ ];
+   *   };
+   * }
+   */
```

Comment:
This doesn't seem like a terribly common idiom for similar functions. I don't expect this will actually stay in this file beyond the near term (I assume this will need to be explained at more length in the nixpkgs manual as it stabilizes), but I did want to leave some breadcrumbs.